torch/distributed/algorithms/ddp_comm_hooks/ddp_zero_hook.py

```diff
@@ -64,58 +71,63 @@ def _broadcast_bucket(
             parameters to broadcast.
         zero (ZeroRedundancyOptimizer): the calling process's
             :class:`ZeroRedundancyOptimizer` instance.
-        assigned_rank (int): the rank assigned to the bucket; it has the
-            updated parameters and serves as the source for the broadcast.
     """
     overlap_info = zero._overlap_info
-    device = overlap_info.params_per_bucket[bucket_index][0].device
-    device_index = zero._device_to_device_index[device]
-    assert bucket_index in zero._buckets[device_index][assigned_rank]
-    overlap_info.broadcast_handles.append(
-        dist.broadcast(
-            zero._buckets[device_index][assigned_rank][bucket_index],
-            src=assigned_rank,
-            async_op=True
-        )
-    )
-
-def _collect_ddp_bucket_info(
+    assert len(overlap_info.assigned_ranks_per_bucket) > bucket_index, \
+        "`assigned_ranks_per_bucket` is not fully constructed"
+    # Sort to ensure the same ordering across ranks
+    assigned_ranks = sorted(overlap_info.assigned_ranks_per_bucket[bucket_index])
```

Comment:
(no need to change anything) curious, do we need to sort this every iteration? I assume this list is static?

Reply:
Currently, I have `overlap_info.assigned_ranks_per_bucket[bucket_index]` as a `set`, which is why I sort every time. We could alternatively make that a `list` and only sort once, but I do not think it makes a big difference either way.